### DISCUSSION TITLE
HMR Doesn't detect changes in files other than root.tsx

### DISCUSSION THREAD

**Discussion by roniaxe** [remix-run/remix#6372] [Q&A]
Hi,

I have enabled `unstable_dev: true` to use the new HMR feature.
However, for some reason, when I update any file other than `root.tsx`, I don't see my UI updates with the changes, unless I refresh the page.

On google console, I can see those logs:
![image](https://github.com/remix-run/remix/assets/19142347/bf7e7eec-7e28-4e8b-bfd5-3351bde5a3eb)

And as you can see, remix detects the correct file that changes, but HMR always logs `[HMR] Component changed [app\root.tsx]` no matter what file has been changed.

I'm using the latest version of remix/react, and I can't see any differences between my app and a new create-remix-app which I tried, and it is working correctly there.

This is my remix.config.js:

```
/** @type {import('@remix-run/dev').AppConfig} */
module.exports = {
    ignoredRouteFiles: ['**/.*'],
    tailwind: true,
    postcss: true,
    server: process.env.NODE_ENV === 'development' ? undefined : './server.ts',
    serverBuildPath: 'api/index.js',
    // When running locally in development mode, we use the built-in remix
    // server. This does not understand the vercel lambda module format,
    // so we default back to the standard build output.
    // server: process.env.NODE_ENV === 'development' ? undefined : './server.js',
    // serverBuildPath: 'api/index.js',
    // appDirectory: "app",
    // assetsBuildDirectory: "public/build",
    // publicPath: "/build/",
    future: {
        v2_meta: true,
        v2_routeConvention: true,
        v2_errorBoundary: true,
        v2_normalizeFormMethod: true,
        unstable_dev: true,
    },
    serverDependenciesToBundle: [/^yet-another-react-lightbox*/],
};

```

I really appreciate some help to fix it.

**Comment by roniaxe** [comment]
@pcattori I saw your video about the new HMR, I was wondering if you can take a look, maybe give me a tip or suggest what I can change to make this work ?

**Comment by roniaxe** [comment]
Update: 
Turns out that it has to do with Radix UI NavigationMenu (https://www.radix-ui.com/docs/primitives/components/navigation-menu).
I'm using their navigation menu as part of my main navigation bar, and when I remove it, HMR is working correctly, and update on any change, but as soon as I bring it back and use it, HMR stop updating the UI on changes. Any Ideas?

**Reply by dimitrisnl** [reply]
Is this still an issue for you? Only this component causes it?
